### PR TITLE
Update Core/ExtraProviders to NETStandard 2.0

### PR DESCRIPTION
**Pull request by phillip-haydon**
Upgrading .Core and .ExtraProviders to NETStandard 2.0

Also downgraded the mvc project to 4.7.2 due to some people being stuck on 4.7.2 for whatever reasons.

### CODE REVIEW

File: Code/Nancy.SimpleAuthentication/SimpleAuthenticationModule.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 namespace Nancy.SimpleAuthentication
 {
-    public class SimpleAuthenticationModule : NancyModule
+    public sealed class SimpleAuthenticationModule : NancyModule
```
Comment (PureKrome) [inline Code/Nancy.SimpleAuthentication/SimpleAuthenticationModule.cs:13]: why `sealed`? (in know what `sealed` keyword is/does). what are you trying to communicate/demonstrate, here?

File: Code/SimpleAuthentication.Core/SimpleAuthentication.Core.csproj [hunk 2]
```diff
@@ -1,156 +1,19 @@
...
+  
+    <ItemGroup>
+        <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
+        <PackageReference Include="RestSharp" Version="106.15.0" />
+        <PackageReference Include="System.Configuration.ConfigurationManager" Version="7.0.0" />
+    </ItemGroup>
+  
+    <ItemGroup>
+      <None Remove="SimpleAuthentication.Core.csproj.DotSettings" />
+    </ItemGroup>
```
Comment (PureKrome) [inline Code/SimpleAuthentication.Core/SimpleAuthentication.Core.csproj:17]: Add this to `gitignore` instead

File: Code/SimpleAuthentication.ExtraProviders/SimpleAuthentication.ExtraProviders.csproj [hunk 3]
```diff
@@ -1,145 +1,19 @@
...
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Import Project="..\..\packages\System.Text.Json.6.0.3\build\System.Text.Json.targets" Condition="Exists('..\..\packages\System.Text.Json.6.0.3\build\System.Text.Json.targets')" />
-  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
     <PropertyGroup>
-      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+        <TargetFramework>netstandard2.0</TargetFramework>
+        <OutputType>Library</OutputType>
+        <LangVersion>latest</LangVersion>
```
Comment (PureKrome) [inline Code/SimpleAuthentication.ExtraProviders/SimpleAuthentication.ExtraProviders.csproj:6]: Don't need this.

File: Code/SimpleAuthentication.Core/SimpleAuthentication.Core.csproj [hunk 4]
```diff
@@ -1,156 +1,19 @@
...
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Import Project="..\..\packages\System.Text.Json.6.0.3\build\System.Text.Json.targets" Condition="Exists('..\..\packages\System.Text.Json.6.0.3\build\System.Text.Json.targets')" />
-  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
     <PropertyGroup>
-      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+        <TargetFramework>netstandard2.0</TargetFramework>
+        <OutputType>Library</OutputType>
+        <LangVersion>latest</LangVersion>
```
Comment (PureKrome) [inline Code/SimpleAuthentication.Core/SimpleAuthentication.Core.csproj:6]: Don't need this. default is sufficient.

File: Code/SimpleAuthentication.Mvc-3/SimpleAuthentication.Mvc-3.csproj [hunk 5]
```diff
@@ -9,11 +9,13 @@
...
     <RootNamespace>SimpleAuthentication.Mvc</RootNamespace>
     <AssemblyName>SimpleAuthentication.Mvc-3</AssemblyName>
-    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <TargetFrameworkProfile />
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
```
Comment (PureKrome) [inline Code/SimpleAuthentication.Mvc-3/SimpleAuthentication.Mvc-3.csproj:18]: Nuke this plz

File: Code/SimpleAuthentication.Mvc/SimpleAuthentication.Mvc.csproj [hunk 6]
```diff
@@ -9,11 +9,13 @@
...
     <RootNamespace>SimpleAuthentication.Mvc</RootNamespace>
     <AssemblyName>SimpleAuthentication.Mvc</AssemblyName>
-    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
     <TargetFrameworkProfile />
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
```
Comment (PureKrome) [inline Code/SimpleAuthentication.Mvc/SimpleAuthentication.Mvc.csproj:18]: 🚮

**Review by PureKrome — CHANGES_REQUESTED** [review]

File: Code/Nancy.SimpleAuthentication/SimpleAuthenticationModule.cs (hunk 1 above)
Comment (phillip-haydon) [reply Code/Nancy.SimpleAuthentication/SimpleAuthenticationModule.cs:13]: It removes the warning on using the virtual `Get` / `Post` methods in the ctor.

Comment (PureKrome) [reply Code/Nancy.SimpleAuthentication/SimpleAuthenticationModule.cs:13]: Ta. TIL.

**Review by PureKrome — APPROVED** [review]

**Pull request merged**